app/controllers/callbacks_controller.rb

```diff
@@ -5,15 +5,18 @@ class CallbacksController < Devise::OmniauthCallbacksController
 
   def github
     auth = request.env['omniauth.auth']
-
     user = User.from_omniauth(auth)
-    user.assign_attributes(token: auth.credentials.token)
-
-    raise MissingMembershipError unless (user.organisations & WHITELISTED_ORGS).any?
 
-    user.save!
-    sign_in_and_redirect user
-    flash[:notice] = 'Signed in successfully.'
+    if (user.organisations & WHITELISTED_ORGS).any?
+      user.save!
+      sign_in_and_redirect user
+      flash[:notice] = 'Signed in successfully.'
+    else
+      redirect_to root_path
+      raise MissingMembershipError
```

Comment:
Raising here has no point — you're just catching the exception two lines below!
At the same time, you'd probably want to raise, since otherwise your page will just render `200 OK` status code